code/modules/overmap/ships/controlled_ship_datum.dm

```diff
@@ -0,0 +1,236 @@
+///Name of the file used for ship name random selection
+#define SHIP_NAMES_FILE "ship_names.json"
+
+/**
+  * # Simulated overmap ship
+  *
+  * A ship that corresponds to an actual, physical shuttle.
+  *
+  * Can be docked to any other overmap datum that has a valid docking process.
+  */
+/datum/overmap/ship/controlled
+	token_type = /obj/overmap/rendered
+	dock_time = 10 SECONDS
+
+	///Vessel estimated thrust
+	var/est_thrust
+	///Average fuel fullness percentage
+	var/avg_fuel_amnt = 100
+	///Cooldown until the ship can be renamed again
+	COOLDOWN_DECLARE(rename_cooldown)
+	///Vessel approximate mass
+	var/mass
+
+	///The docking port of the linked shuttle. To add a port after creating a controlled ship datum, use [/datum/overmap/ship/controlled/proc/connect_new_shuttle_port].
+	VAR_FINAL/obj/docking_port/mobile/shuttle_port
+	///The map template the shuttle was spawned from, if it was indeed created from a template. CAN BE NULL (ex. custom-built ships).
+	var/datum/map_template/shuttle/source_template
+
+	///Assoc list of remaining open job slots (job = remaining slots)
+	var/list/job_slots = list(new /datum/job/captain() = 1, new /datum/job/assistant() = 5)
+	///Manifest list of people on the ship
+	var/list/manifest = list()
+	///Time that next job slot change can occur
+	COOLDOWN_DECLARE(job_slot_adjustment_cooldown)
+	///Whether or not new players are allowed to join the ship
+	var/join_allowed = TRUE
+	///Short memo of the ship shown to new joins
+	var/memo = ""
+
+	///Shipwide bank account used for cargo consoles and bounty payouts.
+	var/datum/bank_account/ship/ship_account
+
+/datum/overmap/ship/controlled/Rename(new_name, force = FALSE)
+	var/oldname = name
+	if(!..() || (!COOLDOWN_FINISHED(src, rename_cooldown) && !force))
+		return FALSE
+	message_admins("[key_name_admin(usr)] renamed vessel '[oldname]' to '[new_name]'")
+	shuttle_port?.name = new_name
```

Comment:
Is there a reason this should ever be null?

Reply:
https://github.com/MarkSuckerberg/Shiptest/blob/8dd1ede1f4a6e25d3b71cc8e657b3e1b51846895/code/game/objects/items/shuttle_creator.dm#L180-182 <- custom shuttles
https://github.com/MarkSuckerberg/Shiptest/blob/eaf6aab68f980bbdf5dd8cbc7213f9abe381af34/code/modules/shuttle/shuttle.dm#L200 <- ruin template/subshuttle loading

Basically, the item is init at first without a template but then immediately given a template. Rename() is called, since it renames other things (like the token, datum itself, does logging) on creation of the datum. I know this isn't a scenario we really want, but it's the most elegant way I can get these two things working (specifically, situations you want to load a ship at a specific dock rather than in transit).